### PR TITLE
Make sure `ext_oneapi_get_default_context` doesn't broke runtime on windows

### PR DESCRIPTION
**Pull request by anmyachev**
Part of #2478 (to reduce diff)

These are quite stable changes, we can merge it without CI on Windows. @gshimansky if you don't mind.

### CODE REVIEW

File: third_party/intel/backend/driver.c [hunk 1]
```diff
@@ -194,8 +194,18 @@ static PyObject *loadBinary(PyObject *self, PyObject *args) {
     const size_t binary_size = PyBytes_Size(py_bytes);
 
     uint8_t *binary_ptr = (uint8_t *)PyBytes_AsString(py_bytes);
-    const auto ctx =
-        sycl_device.get_platform().ext_oneapi_get_default_context();
+    auto platform = sycl_device.get_platform();
```
[the remark this answers is not in the document]
Comment (anmyachev) [reply third_party/intel/backend/driver.c]: Done

> Also, how can the code properly run with no valid context?

This is a explanation for Windows case: https://github.com/intel/intel-xpu-backend-for-triton/pull/2478/files#r1844125901

**Comment by gshimansky** [comment]
I found how pytorch implements the same functionality on Linux and Windows. You may want to take a look.
https://github.com/pytorch/pytorch/blob/main/c10/xpu/XPUFunctions.cpp#L59-L72

**Comment by anmyachev** [comment]
> I found how pytorch implements the same functionality on Linux and Windows. You may want to take a look. https://github.com/pytorch/pytorch/blob/main/c10/xpu/XPUFunctions.cpp#L59-L72

This is interesting, but it will also require us to implement `enumDevices` function https://github.com/pytorch/pytorch/blob/0c7c5d78faa61245700ba6f2d0c237019090f684/c10/xpu/XPUFunctions.cpp#L30. Let's leave the current version for now, since it works, and open an issue on how this can be improved.

**Comment by anmyachev** [comment]
@alexbaden ready for review

**Review by alexbaden — APPROVED** [review]
looks fine for now

**Comment by anmyachev** [comment]
https://github.com/intel/intel-xpu-backend-for-triton/issues/2757

**Pull request merged**